### PR TITLE
[MLv2] Export find-matching-column

### LINKED ISSUE #35675: [MLv2] BE - Expose findMatchingColumn
`findMatchingColumn(columns: ColumnMetadata[], column: ColumnMetadata): ColumnMetadata | null`

If needed, we can pass `query`, `stageIndex` etc

To be used like this
```
findMatchingColumn(filterableColumns(query, stageIndex), breakoutColumn)
```

### PR DESCRIPTION
**Pull request by snoe**
Fixes #35675

### CODE REVIEW

File: src/metabase/lib/js.cljs [hunk 1]
```diff
@@ -996,3 +996,30 @@
...
+   name or `:lib/desired-column-alias` and matched that way.
+   - `query` and `stage-number` are required for this case, since they're needed to resolve the correct name.
+   - Columns with `:id` set are dropped to prevent them matching. (If they didn't match by `:id` above they shouldn't
+   match by name due to a coincidence of column names in different tables.)
+   - String IDs are checked against `:lib/desired-column-alias` first.
+   - If that doesn't match any columns, `:name` is compared next.
+   - The same disambiguation (by `:join-alias` etc.) is applied if there are multiple plausible matches.
+
+   Returns the matching column, or nil if no match is found."
+  [a-ref columns]
```
Comment (ranquild) [inline src/metabase/lib/js.cljs:1024]: We pass a `column`, not ` ref`:

```
[column columns]
(lib.core/find-matching-column (lib.core/ref column) columns))
```

**Review by bshepherdson — APPROVED** [review]
LGTM, but we should figure out calling it with a column or a ref.

**Review by metamben — APPROVED** [review]
It is a bit counterintuitive to look for a column by first converting it into a ref (losing some information), but we can give it a try.

File: src/metabase/lib/js.cljs [hunk 2]
```diff
@@ -997,6 +997,33 @@
...
+   - `query` and `stage-number` are required for this case, since they're needed to resolve the correct name.
+   - Columns with `:id` set are dropped to prevent them matching. (If they didn't match by `:id` above they shouldn't
+   match by name due to a coincidence of column names in different tables.)
+   - String IDs are checked against `:lib/desired-column-alias` first.
+   - If that doesn't match any columns, `:name` is compared next.
+   - The same disambiguation (by `:join-alias` etc.) is applied if there are multiple plausible matches.
+
+   Returns the matching column, or nil if no match is found."
+  [a-query stage-number a-ref columns]
+  (lib.core/find-matching-column a-query stage-number a-ref columns))
```
Comment (ranquild) [inline src/metabase/lib/js.cljs:1025]: nit: rename `a-ref` to `a-ref-or-column`

**Review by ranquild — APPROVED** [review]

**Pull request merged**